model/utils.go

```diff
@@ -218,10 +217,32 @@ type AppError struct {
 	Where         string `json:"-"`                     // The function where it happened in the form of Struct.Func
 	IsOAuth       bool   `json:"is_oauth,omitempty"`    // Whether the error is OAuth specific
 	params        map[string]any
+	wrapped       error
 }
 
 func (er *AppError) Error() string {
-	return er.Where + ": " + er.Message + ", " + er.DetailedError
+	var sb strings.Builder
+
+	// render the error information
+	sb.WriteString(er.Where)
+	sb.WriteString(": ")
+	sb.WriteString(er.Message)
+
+	// only render the detailed error when it's present
+	if er.DetailedError != "" {
+		sb.WriteString(", ")
+		sb.WriteString(er.DetailedError)
+	}
+
+	// render all wrapped errors
+	err := er.wrapped
+	for err != nil {
+		sb.WriteString(", ")
+		sb.WriteString(err.Error())
+		err = errors.Unwrap(err)
+	}
```

Comment:
Can we have a test for this? :)

Reply:
Added a test for the error string rendering.